tests/integration/renderer/test_multi_output.py

```diff
@@ -133,10 +133,8 @@ def set_bc(a):
         dev_tools_hot_reload=False,
     )
 
-    # the UI still renders the output triggered by callback.
-    # The new system does NOT loop infinitely like it used to, each callback
-    # is invoked no more than once.
-    dash_duo.wait_for_text_to_equal("#c", "X")
+    # The new system does NOT trigger callbacks in circular dependencies
+    dash_duo.wait_for_text_to_equal("#c", "")
```

Comment:
This needs to be looked into. Circular dependencies are pruned out entirely now. They used to execute partially on load and unpredictably on prop updates.